libzeth/circuits/circuit_wrapper.hpp

```diff
@@ -25,26 +25,27 @@ template<
     size_t TreeDepth>
 class circuit_wrapper
 {
-private:
-    std::shared_ptr<joinsplit_gadget<
+public:
+    using Field = libff::Fr<ppT>;
+    // Both `joinsplit` and `joinsplit_gadget` are already used in the
+    // namespace.
+    using joinsplit_type = joinsplit_gadget<
         libff::Fr<ppT>,
         HashT,
         HashTreeT,
         NumInputs,
         NumOutputs,
-        TreeDepth>>
-        joinsplit_g;
-
-public:
-    using Field = libff::Fr<ppT>;
+        TreeDepth>;
```

Comment:
Since you use a `using` directive for the Field, why not using it in the `joinsplit_type` directive?

Reply:
switched to use type def